core/src/cluster_info.rs

```diff
@@ -2456,24 +2456,20 @@ impl ClusterInfo {
         thread_pool: &ThreadPool,
         last_print: &mut Instant,
     ) -> Result<()> {
-        let timeout = Duration::new(1, 0);
-        let mut requests = vec![requests_receiver.recv_timeout(timeout)?];
-        let mut num_requests = requests.last().unwrap().packets.len();
-        while let Ok(more_reqs) = requests_receiver.try_recv() {
-            if num_requests >= MAX_GOSSIP_TRAFFIC {
-                continue;
+        const RECV_TIMEOUT: Duration = Duration::from_secs(1);
+        let packets: Vec<_> = requests_receiver.recv_timeout(RECV_TIMEOUT)?.packets.into();
+        let mut packets = VecDeque::from(packets);
+        while let Ok(packet) = requests_receiver.try_recv() {
+            let num_packets = packets.len() + packet.packets.len();
+            if num_packets > MAX_GOSSIP_TRAFFIC {
+                let excess_count = usize::min(MAX_GOSSIP_TRAFFIC - num_packets, packets.len());
```

Comment:
if `num_packets > MAX_GOSSIP_TRAFFIC` then `MAX_GOSSIP_TRAFFIC - num_packets` will overflow, should `excess_count` just be `num_packets - MAX_GOSSIP_TRAFFIC` here?

Reply:
oops, you are right. fixed